Add Bors integration to repository

ebkalderon opened the pull request:
This pull request introduces [Bors](https://bors.tech/) build bot integration to the main Amethyst repository. Benefits of Bors include stateless CI testing and ensuring the stability of the `master` and `develop` branches. I have already installed the Bors application to our GitHub organization, so if this pull request works for us, we can adopt identical changes to all other Amethyst repositories.

Closes #268

Xaeroxe commented on appveyor.yml:
```diff
@@ -1,47 +1,51 @@
+    target: x86_64-pc-windows-msvc
+
+# Allow for occasional compiler breakage on nightly Rust.
 matrix:
```
Error parsing appveyor.yml: (Line: 18, Col: 1, Idx: 406) - (Line: 18, Col: 7, Idx: 412): Duplicate key

Is it possible the indentation on the two lines below this were necessary?  Alternatively I think we can just move `allow_failures` into the matrix key from above.

Xaeroxe commented on appveyor.yml:
```diff
@@ -1,47 +1,50 @@
+    target: x86_64-pc-windows-msvc
+
+  # Allow for occasional compiler breakage on nightly Rust.
+  allow_failures:
```
stick a `-` here

Error parsing appveyor.yml: (Line: 18, Col: 3, Idx: 410) - (Line: 18, Col: 3, Idx: 410): While parsing a block collection, did not find expected '-' indicator.

Xaeroxe commented:
On another note: I'm so glad we aren't using yml for Amethyst configs anymore.

Xaeroxe commented:
Alright let's give this a shot.

@bors r+

ebkalderon commented:
@Xaeroxe The merge will not work because I have placed the *status: working* label on this pull request. 😉 It is not fully ready yet, hang tight!

Xaeroxe commented:
haha sorry I jumped the gun a bit.

ebkalderon commented:
It's okay, it happens!

torkleyy commented on appveyor.yml:
```diff
@@ -1,47 +1,52 @@
+    - staging
+    - trying
+    - develop
+    - master
```
Should also build the latest release branch.

ebkalderon replied:
Ah, good catch. Will add this now.

torkleyy commented:
@Xaeroxe You would write "bors r+" btw.

Xaeroxe commented:
@torkleyy Wouldn't that comment count as an "approval" just now?

torkleyy commented:
No, it has to be on a separate line.

Xaeroxe commented:
Ah, thanks!

Xaeroxe commented:
So what precisely is left to do before this should be merged @ebkalderon ?

ebkalderon commented:
@Xaeroxe I've already done the extra fixes I wanted; I removed the workaround for Cargo in AppVeyor, since the HTTPS issues have already been fixed on AppVeyor's end, I've added appropriate regex to let Bors build all the `release-X.Y` branches, and finally I've documented the `bors.toml` file with comments.

Currently, I'm waiting for all the CI checks to complete. Once they're done, we can merge.

Xaeroxe commented:
bors r+

Xaeroxe commented:
@ebkalderon Do you have access to the queue page for bors?  Sort of like this: https://buildbot2.rust-lang.org/homu/queue/rust I know I'm gonna get anxious in the future waiting to see if bors heard my request or not.

Xaeroxe commented:
Aw :( @ebkalderon Is there any easy way to just add engine-devs as a reviewer?

Xaeroxe commented:
Hey look! https://app.bors.tech/repositories/908/ If you chop off some of that URL you get a nice repository page for us!

Xaeroxe commented:
bors r+

Xaeroxe commented:
bors retry

Xaeroxe commented:
I guess that command doesn't work for this bors.  We should probably increase the time out before merging anyways

Xaeroxe commented:
bors r+
